src/async-wrap.cc

```diff
@@ -282,37 +285,87 @@ bool AsyncWrap::EmitAfter(Environment* env, double async_id) {
 class PromiseWrap : public AsyncWrap {
  public:
   PromiseWrap(Environment* env, Local<Object> object, bool silent)
-    : AsyncWrap(env, object, PROVIDER_PROMISE, silent) {}
+    : AsyncWrap(env, object, PROVIDER_PROMISE, silent) {
+    MakeWeak(this);
```

Comment:
@addaleax , thank you for the explanation. I just read `MakeWeak` API and `node_object_wrap.h`, and I will continue to learn about it.

And I am not quite understand is why we don't need the `MakeWeak` call before this version?